graylog2-web-interface/src/views/logic/search/SearchExecutionState.js

```diff
@@ -6,6 +6,13 @@ import type { QueryString, TimeRange } from '../queries/Query';
 export type GlobalOverride = {
   timerange?: TimeRange,
   query?: QueryString,
+  keep_search_types?: string[],
+  search_types?: {
+    [searchTypeId: string]: {
+      limit: number;
+      offset: number;
+    };
+  }
```

Comment:
Two questions regarding this adjustment of the GlobalOverride type:

- Currently limit and offset can only be used for a message search type. What are the best practices for this case? Should I create a union type?
- I would prefer camelCase for the attribute names. We already talked about he implementation `.toJSON`. So far this only works for top level attributes. How do we deal with these encapsulated attributes? Should I adjust them inside the default `.toJSON()` method as well?